src/bitcoin/wallet.rs

```diff
@@ -84,18 +72,16 @@ impl Wallet {
         }
     }
 
-    pub fn random_transient_sk(&self) -> anyhow::Result<SecretKey> {
-        // TODO: Replace random bytes with SwapId or SharedSwapId?
-        let mut random_bytes = [0u8; 32];
+    pub fn derive_transient_sk(&self, swap_id: SwapId) -> SecretKey {
```

Comment:
I think it'd be simpler and less "make your own" crypto if we were just deriving new keys from the seed under a different path. Let's say '2'.

Reply:
Oh, yeah. I didn't even think we could do that. That's a better approach.